### ISSUE TITLE
Incompatibility with Dockerfile heredoc syntax

### ISSUE THREAD

**Issue by chgl** [bridgecrewio/checkov#5090]
**Describe the issue**

The heredoc syntax in Dockerfiles (https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/) is not parsed correctly by checkov. While a warning is printed, this causes some checks to pass incorrectly.

**Example Value**

The following Dockerfile

```Dockerfile
# syntax=docker/dockerfile:1.4
FROM docker.io/library/ubuntu:22.04

RUN <<EOF
echo "Hello"
echo "World"
mkdir -p /hello/world
# not detected by checkov:
apt update
EOF

# detected by checkov:
RUN apt update
```

Checkov'd via `checkov --file=test.Dockerfile` logs:

```console
2023-05-15 12:44:28,413 [MainThread  ] [WARNI]  An unsupported instruction ECHO was used in test.Dockerfile
2023-05-15 12:44:28,413 [MainThread  ] [WARNI]  An unsupported instruction ECHO was used in test.Dockerfile
2023-05-15 12:44:28,413 [MainThread  ] [WARNI]  An unsupported instruction MKDIR was used in test.Dockerfile
2023-05-15 12:44:28,413 [MainThread  ] [WARNI]  An unsupported instruction APT was used in test.Dockerfile
2023-05-15 12:44:28,413 [MainThread  ] [WARNI]  An unsupported instruction EOF was used in test.Dockerfile
[ dockerfile framework ]: 100%|████████████████████|[1/1], Current File Scanned=test.Dockerfile
[ secrets framework ]: 100%|████████████████████|[1/1], Current File Scanned=test.Dockerfile

       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 2.3.240 

dockerfile scan results:

Passed checks: 34, Failed checks: 4, Skipped checks: 0

Check: CKV_DOCKER_7: "Ensure the base image uses a non latest version tag"
        PASSED for resource: test.Dockerfile.
        File: test.Dockerfile:1-13
        Guide: https://docs.bridgecrew.io/docs/ensure-the-base-image-uses-a-non-latest-version-tag
# ...
Check: CKV_DOCKER_9: "Ensure that APT isn't used"
        FAILED for resource: test.Dockerfile.RUN
        File: test.Dockerfile:13-13
        Guide: https://docs.bridgecrew.io/docs/ensure-docker-apt-is-not-used

                13 | RUN apt update

Check: CKV_DOCKER_5: "Ensure update instructions are not use alone in the Dockerfile"
        FAILED for resource: test.Dockerfile.RUN
        File: test.Dockerfile:13-13
        Guide: https://docs.bridgecrew.io/docs/ensure-update-instructions-are-not-used-alone-in-the-dockerfile

                13 | RUN apt update

Check: CKV_DOCKER_3: "Ensure that a user for the container has been created"
        FAILED for resource: test.Dockerfile.
        File: test.Dockerfile:1-13
        Guide: https://docs.bridgecrew.io/docs/ensure-that-a-user-for-the-container-has-been-created

                1  | # syntax=docker/dockerfile:1.4
                2  | FROM docker.io/library/ubuntu:22.04
                3  | 
                4  | RUN <<EOF
                5  | echo "Hello"
                6  | echo "World"
                7  | mkdir -p /hello/world
                8  | # not detected by checkov:
                9  | apt update
                10 | EOF
                11 | 
                12 | # detected by checkov:
                13 | RUN apt update

Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: test.Dockerfile.
        File: test.Dockerfile:1-13
        Guide: https://docs.bridgecrew.io/docs/ensure-that-healthcheck-instructions-have-been-added-to-container-images

                1  | # syntax=docker/dockerfile:1.4
                2  | FROM docker.io/library/ubuntu:22.04
                3  | 
                4  | RUN <<EOF
                5  | echo "Hello"
                6  | echo "World"
                7  | mkdir -p /hello/world
                8  | # not detected by checkov:
                9  | apt update
                10 | EOF
                11 | 
                12 | # detected by checkov:
                13 | RUN apt update
```

Note that while the `apt update` in L13 is detected, the one inside the heredoc is not.

**Comment by candrews** [comment]
This is still an issue and really should be addressed.
